doc-content/enterprise-only/optimizer/business-optimizer-test-constraints-quarkus-proc.adoc

```diff
@@ -0,0 +1,63 @@
+
+[id='business-optimizer-test-constraints-proc_{CONTEXT}']
+= Test the school timetable constraints
+
+To test each constraint of the timetable project in isolation, use a `ConstraintVerifier` in unit tests. Doing this tests each constraint’s corner cases in isolation from the other tests, which lowers maintenance when adding a new constraint with proper test coverage.
+
+This test verifies that the constraint `TimeTableConstraintProvider::roomConflict`, when given three lessons in the same room and two of the lessons have the same timeslot, penalizes with a match weight of 1. So if the constraint weight is `10hard` it reduces the score by `-10hard`.
```

Comment:
I assume "10hard" is correct and not 2 words?

Reply:
Yes, but good to check.